### ISSUE TITLE
Require all Model Outputs to be Logits

### ISSUE THREAD

**Issue by ramppdev** [autrainer/autrainer#40]
Some models currently allow for the direct application of Softmax or Sigmoid on their outputs (originally aimed at convenience; now basically legacy), e.g.: CNN10:
https://github.com/autrainer/autrainer/blob/d1ca7efbb2a42961ebf4313ef8546e8b1856678a/autrainer/models/cnn_10.py#L11-L19

However all target transforms expect model outputs to be the actual logits leading to a bug where e.g. for multi-label classification we would apply the Sigmoid function twice:
- once in the model when passing `sigmoid_output = True`
- once in the target transform when obtaining the predictions: https://github.com/autrainer/autrainer/blob/d1ca7efbb2a42961ebf4313ef8546e8b1856678a/autrainer/datasets/utils/target_transforms.py#L190-L202

This essentially leads to the issue that all model predictions are wrong, also causing the published [`autrainer/edansa-2019-cnn10-32k-t`](https://huggingface.co/autrainer/edansa-2019-cnn10-32k-t) to be affected.

**Comment by ramppdev** [comment]
> The question is, do we introduce a new method, i.e. use `predict_batch` or `predict_probabilities` to get the sigmoided/softmaxed outputs, and then `postprocess_output` to do what we now do in `predict_batch`?


During inference we have the following pipeline (M = Model, T = target transform): `M.outputs` -> `T.predict_batch` -> `T.decode` (map) -> (decoded).
During output tracking we currently call  `M.outputs` -> `T.predict_batch` -> `T.decode` (map) -> (save dataframe).

We could extend both pipelines to include e.g.  `M.outputs` -> `T.probabilities_batch` -> `T.predict_batch` -> ... with `T.probabilities_batch` applying the following operations:
- (single-target) regression: identity OP; still scaled (no update of `T.predict_batch` required)
- multi-class classification: Softmax (no update of `T.predict_batch` required; we only argmax the outputs)
- multi-label classification: Sigmoid (remove Sigmoid from `T.predict_batch`)

Then in all cases we can save the outputs as well as the result of `T.probabilities_batch` (+ destructure the probabilities to separate columns in the inference `results.csv`).
